dmutils/email/tokens.py

```diff
@@ -100,13 +100,24 @@ def decode_password_reset_token(token, data_api_client):
     try:
         decoded, token_timestamp = decode_token(
             token,
-            current_app.config["SECRET_KEY"],
+            current_app.config["SHARED_EMAIL_KEY"],
             current_app.config["RESET_PASSWORD_SALT"],
             ONE_DAY_IN_SECONDS,
         )
     except fernet.InvalidToken as e:
-        current_app.logger.info("Error changing password: {error}", extra={'error': six.text_type(e)})
-        return {'error': 'token_invalid'}
+        try:
+            # We used to issue tokens with `current_app.config["SECRET_KEY"]` as our key, so we maintain backwards
+            # compatability with them for a short period (1 day) to allow for both type of tokens to be valid until the
+            # older ones have expired and can remove the support for tokens using `current_app.config["SECRET_KEY"]`
+            decoded, token_timestamp = decode_token(
+                token,
+                current_app.config["SECRET_KEY"],
+                current_app.config["RESET_PASSWORD_SALT"],
+                ONE_DAY_IN_SECONDS,
+            )
+        except fernet.InvalidToken as e2:
+            current_app.logger.info("Error changing password: {error}", extra={'error': six.text_type(e2)})
```

Comment:
Is it worth getting rid of this use of `six`?

Reply:
I initially did. However it turns out we are supporting both python 2 and python 3 for utils (I guess it's because they are still being used by the scripts repo which still runs a few things in 2).

Reply:
Fair dos.